lib/shared/puppet_self_service.rb

```diff
@@ -46,4 +46,90 @@ def self.service_running_enabled(name, service = nil)
 
     service_running(name, service) and service_enabled(name, service)
   end
+
+  # Return active postgres resource in Debian
+  # @return [String] pe-postgres fully qualified service is debian
+  def self.pe_postgres_version
+    if Facter.value(:os)['family'].eql?('Debian')
+      "pe-postgresql#{Facter.value(:pe_postgresql_info)['installed_server_version']}"
+    else
+      'pe-postgresql'
+    end
+  end
+
+  # Check if Primary node
+  # @return [Boolean] true is primary node
+  def self.primary?
+    configdir = if Facter.value(:os)['family'].eql?('RedHat') || Facter.value(:os)['family'].eql?('Suse')
```

Comment:
These look pretty good. There is a good amount of repeated code. You could have another method that is responsible for populating `configdir` or to simplify things: how about a method that performs a check for a config file. Something like this (with a better method name and correct syntax)

```
  def self.config_file_exists(configfile)
        configdir = if Facter.value(:os)['family'].eql?('RedHat') || Facter.value(:os)['family'].eql?('Suse')
                  '/etc/sysconfig'
                else
                  '/etc/default'
                end
    File.exist?("#{configdir}/#{configfile}")
  end
```

Then the role functions are like
```
 config_file_exists('pe-puppetserver') && ....
```

Reply:
ok think iv done that, lets see what the acceptance says